rewrite-maven/src/main/java/org/openrewrite/maven/tree/MavenRepository.java

```diff
@@ -63,6 +63,8 @@ public class MavenRepository implements Serializable {
     @Nullable
     String snapshots;
 
+    @EqualsAndHashCode.Include
```

Comment:
I think that this field can be excluded from `equals`/`hashCode`.

Reply:
Would you then want to remove this line again, given the `@EqualsAndHashCode(onlyExplicitlyIncluded = true)` at on the class?
```suggestion
```

Reply:
The issue here is that you can have the same repository defined as a `<mirror>` or as a `<repository>` on a Maven's `settings.xml` file. You can't set the release/snapshot policy in a `<mirror>` configuration.

Reply:
Here's a `settings.xml` that shows the issue:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<settings xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.1.0 http://maven.apache.org/xsd/settings-1.1.0.xsd" xmlns="http://maven.apache.org/SETTINGS/1.1.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <mirrors>
    <mirror>
      <mirrorOf>*,!snapshot-repos</mirrorOf>
      <id>release-repos</id>
      <name>maven-release-repos</name>
      <url>https://my.artifactory/artifactory/maven-release-repos/</url>
    </mirror>
  </mirrors>
  <profiles>
    <profile>
      <id>artifactory</id>
      <repositories>
        <repository>
          <id>release-repos</id>
          <name>maven-release-repos</name>
          <url>https://my.artifactory/artifactory/maven-release-repos/</url>
          <snapshots>
            <enabled>false</enabled>
          </snapshots>
        </repository>
        <repository>
          <id>snapshot-repos</id>
          <name>maven-snapshot-repos</name>
          <url>https://my.artifactory/artifactory/maven-snapshot-repos/</url>
          <releases>
            <enabled>false</enabled>
          </releases>
          <snapshots>
            <enabled>true</enabled>
          </snapshots>
        </repository>
      </repositories>
    </profile>
  </profiles>
  <activeProfiles>
    <activeProfile>artifactory</activeProfile>
  </activeProfiles>
</settings>
```